pkgs/top-level/config.nix

```diff
@@ -235,7 +235,10 @@ let
         `allowUnfree` is enabled. It is not a generic allowlist for all types of licenses.
       '';
       default = [ ];
-      type = types.listOf (types.attrsOf types.anything);
+      # NOTE: The type should be `types.listOf (types.attrsOf types.anything)` -- however, this causes infinite
+      # recursion when `config` is a function for reasons which aren't immediately clear.
+      # For now, we use types.raw, the same type used by `freeformType`.
```

Comment:
It seems clear to me: the available attributes of `pkgs` depend on at least `config.allowAliases`, so `pkgs.lib` cannot be resolved without considering at least some of `config`?